src/base-switch/base-switch.tsx

```diff
@@ -61,13 +66,30 @@ export const BaseSwitch = React.forwardRef<HTMLInputElement, BaseSwitchProps>(
       [setCheckedState],
     );
 
-    const onInputFocus = useCallback(() => {
-      setIsInputFocused(true);
-    }, [setIsInputFocused]);
+    const onInputFocus = useCallback(
+      (e: React.FocusEvent<HTMLInputElement>) => {
+        if (isFocusVisible(e)) {
+          setIsInputFocusedVisible(true);
+        }
+        setIsInputFocused(true);
+      },
+      [setIsInputFocused],
+    );
 
     const onInputBlur = useCallback(() => {
+      if (isInputFocusedVisible) {
+        setIsInputFocusedVisible(false);
+      }
```

Comment:
I wonder if we need this if statement here? isn't it the same?
```suggestion

        setIsInputFocusedVisible(false);
      
```

Reply:
It should be the same yes. Just tested that.